pkg/test/framework/components/istioctl/istioctl.go

```diff
@@ -0,0 +1,90 @@
+//  Copyright 2019 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package istioctl
+
+import (
+	"bytes"
+	"testing"
+
+	"istio.io/istio/istioctl/cmd"
+
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+type Instance interface {
+	// Invoke invokes an istioctl command and returns the output and exception.
+	// Cobra commands don't make it easy to separate stdout and stderr and the string parameter
+	// will receive both.
+	Invoke(args []string) (string, error)
```

Comment:
May I recommend adding strongly typed methods here? It will help maintain all code making use of this component. E.g.:
```go
  Version(remote bool) (string, error)
  VersionOrFail(t, remote bool) string
```
etc.

Reply:
In the old framework there were methods like this in _tests/e2e/framework/istioctl.go_.  However few commands had such methods.  The methods scraped the output.

To keep the work small I'd like to get a minimal framework in, then add strongly typed methods in a follow-up PR.